### PR TITLE
Hotfix26 ingestor metadata

### PR DESCRIPTION
**Pull request by akorosov**
@mortenwh , can you look at this one also?

### CODE REVIEW

File: geospaas/nansat_ingestor/managers.py [hunk 1]
```diff
@@ -3,28 +3,26 @@
...
+                                          DataCenter,
+                                          ISOTopicCategory,
+                                          Location)
+from geospaas.catalog.models import GeographicLocation, DatasetURI, Source, Dataset
 
 class DatasetManager(models.Manager):
     optional_fields = {
-        'entry_id'           : {'nansat_key': 'Entry ID',           'default': uuid.uuid4},
-        'entry_title'        : {'nansat_key': 'Entry Title',        'default': lambda : 'NONE'},
-        'data_center'        : {'nansat_key': 'Data Center',
```
Comment (mortenwh) [inline geospaas/nansat_ingestor/managers.py]: removing this means it must be required by Nansat - do we want that?

File: geospaas/nansat_ingestor/managers.py [hunk 2]
```diff
@@ -3,28 +3,26 @@
...
-                                'default': lambda : DataCenter.objects.get(short_name='NERSC')},
-        'summary'            : {'nansat_key': 'Summary',            'default': lambda : 'NONE'},
-        'ISO_topic_category' : {'nansat_key': 'ISO Topic Category',
```
Comment (mortenwh) [inline geospaas/nansat_ingestor/managers.py]: same here - do we want to require this in Nansat? do we require it?

File: geospaas/nansat_ingestor/managers.py [hunk 3]
```diff
@@ -48,58 +46,86 @@ def get_or_create(self, uri, *args, **kwargs):
...
-            instrument = ii,
-            specs=specs)[0]
+                long_name = instrument['Long_Name'])
+
+        specs = n_metadata.get('specs', '')
+        source, _ = Source.objects.get_or_create(platform=platform,
+                                                 instrument=instrument,
+                                                 specs=specs)
+
+        data_center = json.loads(n_metadata['data_center'])
```
Comment (mortenwh) [inline geospaas/nansat_ingestor/managers.py]: see comment above - this means it must be required in Nansat...

File: geospaas/nansat_ingestor/managers.py [hunk 4]
```diff
@@ -48,58 +46,86 @@ def get_or_create(self, uri, *args, **kwargs):
...
+        data_center = DataCenter.objects.get(
+                bucket_level0=data_center['Bucket_Level0'],
+                bucket_level1=data_center['Bucket_Level1'],
+                bucket_level2=data_center['Bucket_Level2'],
+                bucket_level3=data_center['Bucket_Level3'],
+                short_name=data_center['Short_Name'],
+                long_name=data_center['Long_Name'],
+                data_center_url=data_center['Data_Center_URL'])
+
+        iso_topic_category = json.loads(n_metadata['iso_topic_category'])
```
Comment (mortenwh) [inline geospaas/nansat_ingestor/managers.py]: same here - it should have a default value if it is not required by Nansat

File: geospaas/nansat_ingestor/managers.py [hunk 5]
```diff
@@ -48,58 +46,86 @@ def get_or_create(self, uri, *args, **kwargs):
...
+        iso_topic_category = ISOTopicCategory.objects.get(name=iso_topic_category['iso_topic_category'])
+
+        if 'gcmd_location' in n_metadata:
+            gcmd_location = json.loads(n_metadata['gcmd_location'])
+        else:
+            gcmd_location = pti.get_gcmd_location('SEA SURFACE')
```
Comment (mortenwh) [inline geospaas/nansat_ingestor/managers.py]: I.e., it should rather be added to `optional_fields`

**Comment by akorosov** [comment]
OK, I refactored this code slightly.
Summary:
* default metadata values are provided in two dictionaries for clarity:
`default_char_fields` (id, title, summary) and 
`default_foreign_keys` (data_center, iso_topic_category, location)
* all vocabularies now have get_or_create() method (also with tests) that can create an instance from a dictionary (typically returned by pythesint). This get_or_create is used both in Managers of each vocabulary, and in nansat_ingestor.get_or_create.
* I did not change the default values for summary and title. I think Nansat.__repr__ is too heavy (too much text in information on bands, domain, etc). It obviously needs a better thinking and developing a good convention first (worth a separate issue). 

Conclusion:
the actuall issue is fixed.

@mortenwh , can you check that again?

**Pull request merged**